src/transformers/models/gpt2/modeling_tf_gpt2.py

```diff
@@ -813,32 +813,29 @@ def get_output_embeddings(self):
     def set_output_embeddings(self, value):
         self.set_input_embeddings(value)
 
-    def prepare_inputs_for_generation(self, inputs, past=None, use_cache=None, use_xla=False, **kwargs):
-        # TODO: (Joao) after the TF generator is complete, update GPT2 TF generation to match PT's. NB -- some GPT2
-        # tests will need to be fixed after the change
-
+    def prepare_inputs_for_generation(self, inputs, past=None, use_cache=None, **kwargs):
+        token_type_ids = kwargs.get("token_type_ids", None)
         # only last token for inputs_ids if past is defined in kwargs
         if past:
             inputs = tf.expand_dims(inputs[:, -1], -1)
+            if token_type_ids is not None:
+                token_type_ids = tf.expand_dims(token_type_ids[:, -1], -1)
+
+        position_ids = kwargs.get("position_ids", None)
```

Comment:
I know this is also in PT-GPT2's `prepare_inputs_for_generation`, but I think it is not super perfect here. As we don't do anything for `position_ids` in `_update_model_kwargs_for_generation`, it would have some issue if `position_ids` is provided in `model_kwargs`.

However, it's very unlikely we have use cases where `position_ids` is provided explicitly for `generate`. So there is no urgency at all to address this here.

cc @patrickvonplaten 

p.s. In #17479, `https://github.com/huggingface/transformers/blob/9089b7b95a1b12f19b65872323f13f1f68a6eaa7/src/transformers/models/bart/modeling_tf_bart.py#L1400` is always created.

I believe we can always safely create `position_ids` here for (PT/TF)-GPT2 too ..?